examples/working_with_trait_versions.ipynb

```diff
@@ -0,0 +1,946 @@
+    "    \n",
+    "C++ plugins have the trait view classes (privately) compiled into them, so do not depend on the versions that the host application was built against/ships with. Incompatibilities only become apparent at runtime, when incoming trait data is found to be of a version unsupported by the manager or host."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "75a1877c0c9c3786"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "### Trait IDs\n",
+    "\n",
+    "Each trait encodes its version in its unique ID, up to a maximum version of 999, and zero-padded, such that `kId[:-5]` can be used to strip the version tag."
+   ],
+   "id": "59b7afb066b5bcc9"
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "# An updated trait has a different ID for each version.\n",
+    "assert traits.example.UpdatedTrait_v1.kId == \"openassetio-example:example.Updated.v001\"\n",
+    "assert traits.example.UpdatedTrait_v2.kId == \"openassetio-example:example.Updated.v002\"\n",
+    "\n",
+    "# A newly added trait starts at version 1.\n",
+    "assert traits.example.AddedTrait.kId == \"openassetio-example:example.Added.v001\""
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-05-07T12:30:33.857952Z",
+     "start_time": "2024-05-07T12:30:33.842191Z"
+    }
+   },
+   "id": "a31f8a7263370714",
+   "outputs": [],
+   "execution_count": 3
+  },
+  {
+   "cell_type": "markdown",
+   "source": "This means trait view classes from one version cannot be used to read another version.",
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "6e6d3357806e5f67"
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "from openassetio.trait import TraitsData\n",
+    "\n",
+    "\n",
+    "v1_data = TraitsData()\n",
+    "\n",
+    "traits.example.UpdatedTrait_v1.imbueTo(v1_data)\n",
+    "\n",
+    "assert traits.example.UpdatedTrait.isImbuedTo(v1_data) is False\n",
+    "assert traits.example.UpdatedTrait_v1.isImbuedTo(v1_data) is True\n",
+    "assert traits.example.UpdatedTrait_v2.isImbuedTo(v1_data) is False\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-05-07T12:30:33.870856Z",
+     "start_time": "2024-05-07T12:30:33.858926Z"
+    }
+   },
+   "id": "a147006dd4b9b453",
+   "outputs": [],
+   "execution_count": 4
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Specifications\n",
+    "\n",
+    "Specifications are a way to document well-known sets of traits that categorize entities, relationships, locales, or policies. Agreement on these as an industry is crucial for effective interop.\n",
+    "\n",
+    "However, they do not have an independent version - their version is implicit in the (versioned) traits that they compose. A major consequence of this is that no specification version is embedded in the data itself.\n",
```

Comment:
[question] Does this need updating now they do have a version?

Reply:
Good spot.

Reply:
Updated that paragraph in 0d256674